plugins/dell-dock/fu-dell-dock-i2c-ec.c

```diff
@@ -312,12 +312,24 @@ fu_dell_dock_is_valid_dock (FuDevice *device, GError **error)
 	}
 	result = g_bytes_get_data (data, NULL);
 
-	if (result == NULL || *result != EXPECTED_DOCK_TYPE) {
-		g_set_error_literal (error, FWUPD_ERROR, FWUPD_ERROR_NOT_FOUND,
+	if (result == NULL) {
+		g_set_error_literal (error, FWUPD_ERROR, FWUPD_ERROR_NOT_SUPPORTED,
 				     "No valid dock was found");
 		return FALSE;
 	}
-	return TRUE;
+
+	/* this will trigger setting up all the quirks */
+	if (*result == EXPECTED_DOCK_TYPE) {
```

Comment:
are we sure *result is valid? I'd feel a lot more content getting the size from `g_bytes_get_data` and also checking the `sz >= 1`

Reply:
Right before checked that result wasn't NULL

Reply:
But I guess that's a cleaner style way to check via size. I'll adjust.

Reply:
fixed